packages/app-check/src/internal-api.ts

```diff
@@ -306,9 +326,13 @@ function notifyTokenListeners(
 
   for (const listener of listeners) {
     try {
-      listener(token);
+      listener.listener(token);
```

Comment:
This feels weird with 2 `listener`s. Can you rename the field name to something different.  Perhaps we should just store observers in `state. tokenListeners` instead of `AppCheckTokenListenerInternal`s.

Reply:
I converted these to observers.